### PR TITLE
Migrate SMTPConnectionForm to Form Provider

### LINKED ISSUE #41593: `SMTPConnectionForm`

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/41593

### Description
Migrate SMTP Connection form away from settings batch form and to using the Form Provider. 

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Go to Admin -> Settings -> Email
2. The connection form should be updatable. Validation should still work, and should be able to send test emails

### Checklist
- [x] Form Validation
- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx [hunk 1]
```diff
@@ -113,50 +139,140 @@ export const SMTPConnectionForm = ({
...
-                >
-                  {SEND_TEST_BUTTON_STATES[sendingEmail]}
-                </Button>
-              ) : null}
-              <Button
-                className={CS.mr1}
-                disabled={disabled}
-                onClick={handleClearEmailSettings}
+    <Flex justify="space-between">
+      <Stack spacing="sm" maw={400} pl="0.5rem">
```
Comment (rafpaf) [inline frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx:143]: Maybe `style={{paddingInlineStart: "0.5rem"}}`

File: frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx [hunk 2]
```diff
@@ -113,50 +139,140 @@ export const SMTPConnectionForm = ({
...
+                  mb="1rem"
+                >
+                  {testEmailError}
+                </Text>
+              )}
+              <Box mt="1rem">
+                <FormSubmitButton
+                  label={t`Save changes`}
+                  disabled={!dirty}
+                  mr="1.5rem"
```
Comment (rafpaf) [inline frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx:256]: marginInlineEnd could go here I think

Comment (rafpaf) [reply frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx:256]: Or perhaps it's neater to use flexbox

Comment (npfitz) [reply frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx:256]: I briefly threw them in a flex box but it wanted to put them on 2 lines haha. I could try again though (I didn't try very hard)

**Review by rafpaf — APPROVED** [review]
hot

**Pull request merged**